### PR TITLE
Support c string literals

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

File: crates/hir-ty/src/mir/lower.rs [hunk 1]
```diff
@@ -1112,15 +1112,24 @@ impl<'ctx> MirLowerCtx<'ctx> {
...
                 data.extend(b.len().to_le_bytes());
                 let mut mm = MemoryMap::default();
                 mm.insert(0, b.to_vec());
                 return Ok(Operand::from_concrete_const(data, mm, ty));
             }
+            hir_def::hir::Literal::CString(b) => {
+                let b = b.as_bytes();
+                let mut data = Vec::with_capacity(mem::size_of::<usize>() * 2);
+                data.extend(0usize.to_le_bytes());
+                data.extend(b.len().to_le_bytes());
```
Comment (HKalbasi) [inline crates/hir-ty/src/mir/lower.rs:1126]: Shouldn't this be `b.len() + 1`? (I'm not too familiar with `CStr`, just wanted to double check)

Comment (Veykril) [reply crates/hir-ty/src/mir/lower.rs:1126]: It should!

File: crates/hir-ty/src/infer/expr.rs [hunk 2]
```diff
@@ -815,7 +815,7 @@ impl<'a> InferenceContext<'a> {
             Expr::Array(array) => self.infer_expr_array(array, expected),
             Expr::Literal(lit) => match lit {
                 Literal::Bool(..) => self.result.standard_types.bool_.clone(),
-                Literal::String(..) => {
+                Literal::String(..) | Literal::CString(..) => {
                     TyKind::Ref(Mutability::Not, static_lifetime(), TyKind::Str.intern(Interner))
                         .intern(Interner)
```
Comment (lowr) [inline crates/hir-ty/src/infer/expr.rs:820]: The type of c string literals should be `&CStr`, for which [a new lang item `CStr`][lang-item] has been introduced. Also, some hir-ty tests would be nice!

[lang-item]: https://github.com/rust-lang/rust/blob/77c836e1ae582661924d3b6ec4d57a2de120f59f/library/core/src/ffi/c_str.rs#L84

Comment (Veykril) [reply crates/hir-ty/src/infer/expr.rs:820]: Ah, right. For some reason I completely forgot about that 😅

**Pull request merged**